PhysicsTools/NanoAOD/python/jetsAK8_cff.py

```diff
@@ -317,10 +317,50 @@ def nanoAOD_addDeepInfoAK8(process, addDeepBTag, addDeepBoostedJet, addDeepDoubl
 fatJetTable.variables.pt.precision=10
 subJetTable.variables.pt.precision=10
 
+##############################################################
+# AK8 constituents
+###############################################################
+finalJetsAK8PFConstituents = cms.EDProducer("PatJetConstituentPtrSelector",
+    src = cms.InputTag("finalJetsAK8"),
+    cut = cms.string("abs(eta) <= 2.5")
+)
+
+finalJetsPFConstituents = cms.EDProducer("PackedCandidatePtrMerger",
+    src = cms.VInputTag(cms.InputTag("finalJetsAK8PFConstituents", "constituents")),
+    skipNulls = cms.bool(True),
+    warnOnSkip = cms.bool(True)
+)
+
+pfCandidatesTable = cms.EDProducer("SimplePATCandidateFlatTableProducer",
+    src = cms.InputTag("finalJetsPFConstituents"),
+    cut = cms.string(""),
+    name = cms.string("PFCand"),
+    doc = cms.string("PF candidate constituents of AK8 puppi jets (FatJet) with |eta| <= 2.5."),
+    singleton = cms.bool(False),
+    extension = cms.bool(False),
+    variables = cms.PSet(
+        pt = Var("pt * puppiWeight()", float, doc="Puppi-weighted pt", precision=10),
+        mass = Var("mass * puppiWeight()", float, doc="Puppi-weighted mass", precision=10),
+        eta = Var("eta", float, precision=12),
+        phi = Var("phi", float, precision=12),
+        pdgId  = Var("pdgId", int, doc="PF candidate type (+/-211 = ChgHad, 130 = NeuHad, 22 = Photon, +/-11 = Electron, +/-13 = Muon, 1 = HFHad, 2 = HFEM)")
+    )
+)
+
+finalJetsAK8ConstituentsTable = cms.EDProducer("SimplePatJetConstituentTableProducer",
+    name = cms.string(fatJetTable.name.value()+"PFCand"),
+    candIdxName = cms.string("PFCandIdx"),
+    candIdxDoc = cms.string("Index in the PFCand table"),
+    jets = fatJetTable.src,
+    candidates = pfCandidatesTable.src,
+    jetCut = cms.string("")
```

Comment:
Would it be better to use
```suggestion
    jetCut = fatJetTable.cut
```
to ensure consistency? It does not matter now as both `finalJetsAK8` and `fatJetTable` apply `pt > 170`, but just to be on the safe side.

Reply:
I will correct this also.